src/ort_genai_c.cpp

```diff
@@ -372,13 +372,13 @@ OgaResult* OGA_API_CALL OgaGenerator_GetOutput(const OgaGenerator* oga_generator
 
 OgaResult* OGA_API_CALL OgaGenerator_GetLogits(OgaGenerator* oga_generator, OgaTensor** out) {
   OGA_TRY
-  auto& generator = *reinterpret_cast<Generators::Generator*>(oga_generator);
-  auto logits_span = generator.GetLogits();
-  const std::array<int64_t, 3> shape{generator.state_->params_->search.batch_size, 1, generator.model_->config_->model.vocab_size};
+  auto generator = reinterpret_cast<Generators::Generator*>(oga_generator);
```

Comment:
Just curious why you switched to generator being a pointer vs a reference?

Reply:
Just to be consistent with SetLogits(), which do you prefer?

Reply:
Slight preference for references when the value can't be nullptr, but it's up to you. There's plenty of other pointers there.